shaders/lighting.frag

```diff
@@ -90,19 +95,47 @@ void main()
         Lo += (kD * albedo / PI + specular) * scene.directionalLight.color.rgb * NoL;
     }
 
+    // Point Light Calculations
+    for (int i = 0; i < pointLights.count; i++)
+    {
+        PointLight light = pointLights.lights[i];
+        vec3 lightPos = light.position.xyz;
+        vec3 L = normalize(lightPos - position);
+        float distance = length(lightPos - position);
+        float attenuation = clamp(1.0 - (distance / light.range), 0.0, 1.0);
+
+        vec3 H = normalize(V + L);
+        vec3 F = FresnelSchlick(max(dot(H, V), 0.0), F0);
+
+        float NDF = DistributionGGX(N, H, roughness);
```

Comment:
from what i can tell a lot of this code is duplicated from the directional light calculation. I think we can generalize this to a function, right?

Reply:
I refactored the lighting shader. Much cleaner now in my opinion. But let me know what you think of it.